es5-persistence/src/main/java/com/netflix/conductor/elasticsearch/query/parser/ComparisonOp.java

```diff
@@ -70,7 +62,7 @@ protected void _parse() throws Exception {
 			this.value = "IS";
 		}else if(peeked[0] == '!' && peeked[1] == '='){
 			this.value = "!=";
-		}else if(peeked.length == betwnLen && new String(peeked).equals(Operators.BETWEEN.value())) {
+		}else if(peeked[0] == 'B' && peeked[1] == 'E' && peeked[2] == 'T' && peeked[3] == 'W' && peeked[4] == 'E' && peeked[5] == 'E' && peeked[6] == 'N'){
```

Comment:
Couple of things:
1. Now that we're not checking length of peeked, could it possibly throw ArrayIndexOutOfBounds exception?
2. Given that `STARTS_WITH` below still checks with String equals, what was the case for checking each character by element?

Reply:
Ahh good catch on the `ArrayIndexOutOfBounds` exception. Pushed out a fix for this.

Reply:
The parse method always peeks the `maxOperatorLength` number of characters. After the `STARTS_WITH` operator was added, the length of this operator is now the max length (11). Now consider the search query excerpt `startTime BETWEEN 100 AND 200`. In this case, the parse method would now parse the following string `BETWEEN 100` which is of length 11. This would fail the String equals comparison and the BETWEEN operator is now non-functional.